### PR TITLE
refactor search dao and remove unused routes

### PR DESCRIPTION
**Pull request by camelliazhang**
* Remove unused routes related to metrics/flows/comments, etc
* Remove advSearch
* Move the SearchDAO into Wherehows-dao to prepare the next refactor
* Play component in build.gradle will be removed in a follow up pull request

### CODE REVIEW

File: wherehows-dao/build.gradle [hunk 1]
```diff
@@ -15,6 +15,11 @@ dependencies {
   runtime externalDependency.hikaricp
 
   testCompile externalDependency.testng
+
+  compile externalDependency.play_java_ws
+  compile externalDependency.play_filter
```
Comment (alyiwang) [inline wherehows-dao/build.gradle]: probably don't need cache and filter?

**Comment by camelliazhang** [comment]
Removed the following dependency as well:
-  compile externalDependency.play_filter
-  compile externalDependency.play_cache

**Comment by theseyi** [comment]
Coverage decreased  @camelliazhang ?

**Comment by camelliazhang** [comment]
@theseyi , the refactor is mainly on removing and migrating, the code coverage may not be correctly calculated?

**Pull request merged**